packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -604,6 +664,19 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		this.revertibles.delete(revertible);
 	}
 
+	private disposeRevertibleRoots(revision: RevisionTag): void {
+		const roots = this.removedRoots.getRoots(revision);
+		if (roots.length > 0) {
+			this.withCombinedVisitor((visitor) => {
+				// get all the roots associated with the revision
+				roots
+					.map((root) => this.removedRoots.toFieldKey(root))
```

Comment:
I recommend doing the mapping as part of the forEach, otherwise you'll be causing a whole additional array to be created.